### ISSUE TITLE
508-defect-3 [COGNITION]: BAM2 Debt Letters - The debt download table should not have sorting icons if they're not actionable

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#16789]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Jennifer_

## User Story or Problem Statement

As a screen reader user, I am confused when the download debt letters table headers announce themselves as "clickable", but when I press Enter or Space, nothing happens.

## Details

The download debt letters table columns 1 and 2 have icons that would suggest the columns are sortable. If these will not be sortable, these icons should be removed. Screenshot attached below.

## Acceptance Criteria

- [ ] Table columns are either wired to be sortable or the icons are removed
- [ ] If the table is made sortable, it's done so in an accessible manner
- [ ] Any sorting function has been discussed and collaborated with the accessibility specialist

## Environment

* https://staging.va.gov/manage-va-debt/your-debt/debt-letters
* Test user 38

## WCAG or Vendor Guidance (optional)

* [Name, Role, Value: Understanding SC 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)

## Screenshots or Trace Logs

![Screen Shot 2020-12-01 at 4 59 53 PM](https://user-images.githubusercontent.com/934879/100807752-799a6680-33f8-11eb-9f68-e99f8cce0eaf.png)

**Comment by bradl3yC** [comment]
@1Copenut The onClick action is encapsulated in the <th>.  You probably didn't notice a change in the sorting because the data is exactly the same between both of those elements

**Comment by 1Copenut** [comment]
@bradl3yC Ah, thank you for the clarification. If there's a test user we can spot check that has a variety of debt letters, I'll give it another quick look and revise / close as appropriate.

**Comment by bradl3yC** [comment]
@1Copenut I don't think we do and our BE engineer is on PTO - You could probably check with dev tools as it will show the shift (their keys/ids should trade places)

**Comment by bradl3yC** [comment]
@1Copenut maybe this will suffice? I did this locally 
![filter](https://user-images.githubusercontent.com/6165421/100931268-4ae3c500-34b8-11eb-85af-4e3812949333.gif)

**Comment by bradl3yC** [comment]
@1Copenut Ready for validation

**Comment by 1Copenut** [comment]
@bradl3yC I took another look in staging, and wasn't able to verify this behavior. I tried clicking on the table headings, and tried to tab to them. The click events didn't trigger the sort, and there was no element to take keyboard focus, so it's not fully accessible yet. Let's chat tomorrow about this one.

**Comment by bradl3yC** [comment]
@1Copenut we're waiting on the DMC to change up some data so we can see the table shift.. I made the clickable headings also focusable so you should see that in staging now.  Hopefully the data will be updated soon

**Comment by bradl3yC** [comment]
@1Copenut the DMC updated their data, this is ready to test

**Comment by 1Copenut** [comment]
@bradl3yC Should I be looking at this issue in Staging? I'm still seeing "TOP" for test user 1's Chapter 35 education debt block from the original description.

**Comment by 1Copenut** [comment]
@bradl3yC & @leahkeeler I verified the sorting is now working. I'll go ahead and close this issue, and open a related one for the sorting accessibility. Keyboard and screen reader users can't sort the table because of negative tab index, but that is outside the scope of this original ask.
